src/ray/object_manager/plasma/client.cc

```diff
@@ -702,6 +711,10 @@ Status PlasmaClient::Impl::MarkObjectUnused(const ObjectID &object_id) {
 Status PlasmaClient::Impl::Release(const ObjectID &object_id) {
```

Comment:
I think if I remember correctly, we currently pin the plasma buffer for mutable objects, by holding a reference here [class](https://github.com/ray-project/ray/blob/e710b81d31cef47c9c162c3caaff3f5dd25db47c/src/ray/core_worker/experimental_mutable_object_manager.h#L60). This is to make sure that the object doesn't get GCed by the plasma store. So I don't think we want to piggy back on the normal buffer destructor path. Instead, we should create our own shared pointer class to populate the result returned by this [method](https://github.com/ray-project/ray/blob/e710b81d31cef47c9c162c3caaff3f5dd25db47c/src/ray/core_worker/experimental_mutable_object_manager.cc#L258) and call the callback on the destruction of that pointer.

Reply:
Fixed.